Clinical scheduler UI

rlorenzo opened the pull request:
Still in draft, but read-only mode is working and ready for review

bsedwards commented on web/Areas/ClinicalScheduler/Services/PersonService.cs:
```diff
@@ -0,0 +1,253 @@
+    /// Service for handling person and clinician data from Clinical Scheduler context
+    /// Replaces direct AAUD context access by using person data available in Clinical Scheduler views
+    /// </summary>
+    public class PersonService : BaseClinicalSchedulerService
```
Called PersonService but is focused on clinicians.

I don't see any permission checking.

bsedwards commented on web/Areas/ClinicalScheduler/Services/PersonService.cs:
```diff
@@ -0,0 +1,253 @@
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns>List of unique MothraIds</returns>
+        public async Task<List<string>> GetAllMothraIdsAsync(CancellationToken cancellationToken = default)
```
Needed?

bsedwards commented on web/Areas/ClinicalScheduler/Services/RotationService.cs:
```diff
@@ -0,0 +1,287 @@
+                    .AsNoTracking()
+                    .Include(r => r.Service);
+
+                // Note: activeOnly parameter is kept for future implementation
```
There definitely is an active flag on the table, once you update the mapping in the context class (see comment there.)

bsedwards commented on web/Areas/ClinicalScheduler/Services/RotationService.cs:
```diff
@@ -0,0 +1,287 @@
+        /// <param name="activeOnly">If true, filters to only active rotations (placeholder for future filtering)</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns>List of rotations with their associated service data</returns>
+        public async Task<List<Rotation>> GetRotationsAsync(bool activeOnly = true, CancellationToken cancellationToken = default)
```
Not sure what the use for the cancellation token is

rlorenzo replied:
It is best practices to include a cancellation token in async methods. It is used to allow canceling long running tasks, like database queries.

bsedwards commented on web/Areas/ClinicalScheduler/Services/WeekService.cs:
```diff
@@ -0,0 +1,238 @@
+                    .OrderBy(wgy => wgy.WeekNum)
+                    .Select(wgy => new VWeek
+                    {
+                        WeekId = wgy.WeekId,
```
A lot of boilerplate, but maybe that can be removed when you update the context class mapping

bsedwards commented on web/Classes/SQLContext/ClinicalSchedulerContext.cs:
```diff
@@ -0,0 +1,195 @@
+    {
+    }
+
+    public virtual DbSet<Rotation> Rotations { get; set; }
```
I see that you're using the CTS objects, but that's not ideal. Those objects are used by CTS and are based on views (because CTS data is in the VIPER database, and clinical scheduler data is still in the clinicalscheduler database, CTS has to use views defined in the VIPER database).

Your clinical scheduler context should map to the tables in the clinical scheduler DB. You'll need to create new objects e.g. in web.Models.ClinicalScheduler

bsedwards commented on web/Classes/TermCodes.cs:
```diff
@@ -0,0 +1,80 @@
+namespace Viper.Classes
```
Maybe you can find add what you need to areas\curriculum\services\termcodeservice.cs?

At least there you can see how term codes are translated to the term name.

The DVM students are semester based, so they use 02 (Spring) 04 (Summer) and 09 (Fall)

bsedwards commented on web/Models/CTS/ScheduleAudit.cs:
```diff
@@ -0,0 +1,22 @@
+using Viper.Models.VIPER;
```
These should go in ClinicalScheduler, not CTS

rlorenzo commented on web/Areas/ClinicalScheduler/Controllers/CliniciansController.cs:
```diff
@@ -0,0 +1,587 @@
+                _logger.LogInformation("Found {RotationCount} unique rotations for clinician {MothraId}", rotations.Count, mothraId);
+                return Ok(rotations);
+            }
+            catch (Exception ex)
```
Where are we sending email notifications?

rlorenzo replied:
It looks like any log that is error+ will get emailed out.

rlorenzo commented on VueApp/src/ClinicalScheduler/pages/RotationScheduleView.vue:
```diff
@@ -0,0 +1,808 @@
+        const year = date.getFullYear()
+
+        // Find which season this week belongs to
+        const season = seasons.find(s => s.months.includes(month))
```
Moving logic to the backend

rlorenzo commented on web/appsettings.json:
```diff
@@ -1,21 +1,28 @@
+    },
+    "EmailNotifications": {
+        "PrimaryEvaluatorRemoved": {
+            "To": ["lgissvoran@ucdavis.edu"],
```
@bsedwards, I am unsure if it is okay to put the email address in here on a public repo. Is there a way to inject the config into the Jenkins build?

bsedwards commented:
> Is there a test email that I can redirect messages to on TEST?

In the past, we've detected the environment and sent messages to either a developer, or the logged in user, depending on the functionality being tested.

bsedwards commented:
Primary evaluator removed email looks good. I think Primary evaluator changed is unneeded - as long as a primary exists Lauren is not concerned.

rlorenzo commented:
@bsedwards Removed the email when the primary evaluator is replaced.

Added the missing permissions check for the Rotations controller and added additional tests to cover this. Also improved the rotation permissions check.
